content/academy/apify_platform/first_actor/actor_building.md

```diff
@@ -0,0 +1,516 @@
+
+    for (const offer of $('#aod-offer')) {
+        const element = $(offer);
+
+        promises.push(
+            Apify.pushData({
+                ...data,
+                sellerName: element.find('div[id*="soldBy"] a[aria-label]').text().trim(),
+                offer: element.find('.a-price .a-offscreen').text().trim(),
+            }),
+        );
+    }
+
+    return Promise.all(promises);
+};
+```
+
+If Amazon hasn't changed any of their selectors or significantly updated any of their [anti-scraping measures]({{@link anti_scraping.md}}), the results in your dataset should each look like this:
+
+```JSON
+{
+  "title": "Apple iPhone XR, 64GB, Black - Unlocked (Renewed)",
+  "asin": "B07P6Y7954",
+  "itemUrl": "https://www.amazon.com/Apple-iPhone-XR-Fully-Unlocked/dp/B07P6Y7954/ref=sr_1_3?keywords=iphone&qid=1649677535&sr=8-3",
+  "keyword": "iphone",
+  "description": "With the iPhone XR you get a roomy 6.1-inch display, fast enough performance from Apple's A12 Bionic processor and good camera quality in a colorful design and affordable package. Apple has included the all-new Liquid Retina LCD as the display on the iPhone XR. Apple released the iPhone XR with a smattering of color options. Both the glass back and the metal frame are brightly colored, with the glass using an in-depth seven-layer color process to achieve the rich finish and the Apple-exclusive aluminum alloy anodized to match. Instead of 3D Touch, the iPhone XR replicates the experience through \"Haptic Touch\". Advanced Face ID lets you securely unlock your iPhone and log in to apps with just a glance.",
+  "sellerName": "Cell Surfers",
+  "offer": "$249.97"
+}
+```
+
+## [](#calling-another-actor) Calling another actor
+
+If you remember from our project's requirements outlined in the previous lesson, once the crawler has finished running, we have to email ourselves a public link to the dataset by using a [public actor which sends emails](https://console.apify.com/actors/e643gqfZae2TfQEbA/?addFromActorId=e643gqfZae2TfQEbA#/console). Luckily, the ability to do this programmatically is avaialable right within the Apify SDK with the [`Apify.call()`](https://sdk.apify.com/docs/api/apify#apifycallactid-input-options) function.
+
+Let's add a bit of code to the end of our actor to send this email:
+
+```JavaScript
+log.info('Starting the crawl.');
+await crawler.run();
+log.info('Crawl finished.');
+
+log.info('Sending dataset link...');
+const dataset = await Apify.openDataset();
+const { id } = await dataset.getInfo();
+
+await Apify.call('apify/send-mail', {
+    to: 'YOUR_EMAIL@gmail.com',
+    subject: 'Amazon Dataset',
+    text: `Dataset link: https://api.apify.com/v2/datasets/${id}/items?clean=true&format=json`,
+});
+
+log.info('Finished.');
+```
+
+> Refer to <a href="https://docs.apify.com/api/v2#/reference/datasets/dataset/get-dataset" target="_blank">this section</a> of the API documentation to understand how we came up with this public dataset URL
```

Comment:
Please, use a regular markdown link here. If you do wanna use HTML links with `target="_blank"`, then please add `rel="noopener noreferrer"` so we don't expose ourselves to any shenanigans. But better to just use markdown links - less typing 💪

Reply:
@mstephen19 We used HTML links in the past, because the system did not open markdown links in new tabs. Now it does, so there's no need to use HTML links anywhere and if you see a HTML link without custom props, please change it to markdown links as well.

Reply:
Changed all `<a>` tags to md links